### PR TITLE
Fix query

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: src/app/services/data-table.service.ts [hunk 1]
```diff
@@ -117,7 +123,9 @@ export class DataTableService {
   }
 
   set dataTable(value: DataTable) {
-    this._dataTable = value;
+    if (value instanceof DataTable) {
```
Comment (gijskant) [inline src/app/services/data-table.service.ts]: In what situations is something passed that is not an instance of `DataTable`?

Comment (beaugogh) [reply src/app/services/data-table.service.ts]: seems to be related to zone.js regressions, but not entirely sure, no point in the our code that assigns values other than DataTable, but while chaining promises, boolean values somehow got assigned to data table

File: src/app/services/query.service.ts [hunk 2]
```diff
@@ -445,72 +474,94 @@ export class QueryService {
...
+        this.isLoadingSubjectCount_2 = true;
+        this.isLoadingObservationCount_2 = true;
+
+        this.query = null; // clear query
+        // update the subject count and observation count in the 2nd step
+        const constraint_1_2: Constraint = this.constraintService.constraint_1_2();
+        this.resourceService.getCounts(constraint_1_2)
+          .subscribe(
+            (countResponse) => {
+              const index = this.queueOfCalls_2.indexOf(timeStamp.getMilliseconds());
```
Comment (gijskant) [inline src/app/services/query.service.ts]: Is this queue of calls really needed?

Comment (beaugogh) [reply src/app/services/query.service.ts]: there is still some parallel calls, not every call is chained

**Pull request merged**